### PR TITLE
Show commission fee at Earn page

### PR DESCRIPTION
**Pull request by simison**
Adding mention of commission + Stripe fee at Earn page on couple places.

- Related to this, see also aligning Jetpack plans in the backend D112347-code
- Copy isn't the best, but I used what already existed so that I don't have to wait for translations. Next up I'll improve the copy.
- I linked to the support doc where commissions across plans is listed. We should link to plans grid from the docs (TODO)
    Screenshot 2023-06-01 at 15 00 02

- Plans grid doesn't have the commission shown across plans yet; once that's there let's swap link to plans grid instead of docs. That way this works as an upsell to lower the fee. Copy could then be for example:

    > With your current plan, the transaction fee for payments is 10% (+ Stripe fees). Upgrade to lower it.

## Proposed Changes

* Adds blurb about commission fee at Earn page. When the page will be redesigned, we'll include this info then too.

## Testing Instructions

* You should have Stripe disconnected, start with free site
* Go to /earn
* See fees for your plan — try updating to a plan and refresh. You should see fees also for Jetpack sites.
 
Screenshot 2023-06-02 at 13 11 54

* Press "Enable payments button" and see Stripe connection intro, note fees again

Screenshot 2023-06-02 at 12 38 07

* After connecting to Stripe, you'll see fees again. This was already here so no changes to this.

Screenshot 2023-06-02 at 13 09 37

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/my-sites/earn/memberships/index.jsx [hunk 1]
```diff
@@ -697,6 +706,8 @@ const mapStateToProps = ( state ) => {
 	const siteId = getSelectedSiteId( state );
 	const earnings = getEarningsWithDefaultsForSiteId( state, siteId );
 
+	console.log( 'memberships:', earnings ); // eslint-disable-line no-console
```
Comment (lezama) [inline client/my-sites/earn/memberships/index.jsx]: yolo

File: client/my-sites/earn/home.tsx [hunk 2]
```diff
@@ -512,24 +525,30 @@ const Home: FunctionComponent< ConnectedProps > = ( {
...
-		</Fragment>
+		</>
 	);
 };
 
 export default connect(
 	( state: IAppState ) => {
 		// Default value of 0 to appease TypeScript for selectors that don't allow a null site ID value.
-		const siteId = getSelectedSiteId( state ) ?? 0;
+		//const siteId = getSelectedSiteId( state ) ?? 0;
```
Comment (lezama) [inline client/my-sites/earn/home.tsx]: should we keep this comment?

File: client/my-sites/earn/home.tsx [hunk 3]
```diff
@@ -512,24 +525,30 @@ const Home: FunctionComponent< ConnectedProps > = ( {
...
+		const earnings = getEarningsWithDefaultsForSiteId( state, siteId );
 
 		const hasConnectedAccount =
 			state?.memberships?.settings?.[ siteId ]?.connectedAccountId ?? null;
 		const sitePlanSlug = getSitePlanSlug( state, siteId );
 		const hasWordAdsFeature = siteHasWordAds( state, siteId );
 		const isLoading = hasConnectedAccount === null || sitePlanSlug === null;
 		const isJetpack = isJetpackSite( state, siteId );
+
+		console.log( 'home:', earnings ); // eslint-disable-line no-console
```
Comment (lezama) [inline client/my-sites/earn/home.tsx]: console.log

File: client/state/memberships/earnings/selectors.js [hunk 4]
```diff
@@ -8,6 +8,7 @@ export function getEarningsForSiteId( state, siteId ) {
 
 export function getEarningsWithDefaultsForSiteId( state, siteId ) {
 	const earnings = getEarningsForSiteId( state, siteId ) ?? {};
+	console.log( 'getEarningsWithDefaultsForSiteId', earnings, siteId ); // eslint-disable-line no-console
```
Comment (lezama) [inline client/state/memberships/earnings/selectors.js]: 👋

File: client/my-sites/earn/home.tsx (hunk 2 above)
Comment (simison) [reply client/my-sites/earn/home.tsx]: Ah no, I was debugging something.

**Review by lezama — APPROVED** [review]
Looking good!

File: client/my-sites/earn/components/commission-fees.tsx [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+	commission: number;
+};
+
+const CommissionFees = ( { commission, className }: CommissionFeesProps ) => {
+	const translate = useTranslate();
+
+	return (
+		commission !== null && (
+			<div className={ className }>
+				{ translate(
```
Comment (millerf) [inline client/my-sites/earn/components/commission-fees.tsx]: I am not sure about that translation. I would leave a translation comment.
Also, stripe: '2.9%+30c' cannot be localised, which is an issue I think...

File: client/my-sites/earn/components/commission-fees.tsx [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+					'On your current plan, WordPress.com charges {{em}}%(commission)s{{/em}}.{{br/}} Additionally, Stripe charges are typically %(stripe)s. {{a}}Learn more{{/a}}',
+					{
+						args: {
+							commission: '' + parseFloat( commission ) * 100 + '%',
+							stripe: '2.9%+30c',
+						},
+						components: {
```
Comment (millerf) [inline client/my-sites/earn/components/commission-fees.tsx:25]: I think `createInterpolateElement(` can be used. But I am not sure which one is closer to the latest standards....
It's a bit more readable. Cf something like [this](https://github.com/Automattic/wp-calypso/blob/dc483dabe82a9c97bb34f441e2f359a6d68ec32d/client/blocks/import-light/summary/index.tsx#L25)

File: client/my-sites/earn/style.scss [hunk 7]
```diff
@@ -38,3 +38,15 @@ See https://dequeuniversity.com/rules/axe/4.6/link-in-text-block?application=Axe
...
 }
+
+.earn__notes {
+	clear: both;
+	padding-bottom: 12px;
+	padding-top: 24px;
+	font-size: $font-body-small;
+	line-height: 1.6;
+	font-style: italic;
+}
```
Comment (millerf) [inline client/my-sites/earn/style.scss]: I would take advantage of scss and use something like:
```
.earn__notes {
	clear: both;
	padding-bottom: 12px;
	padding-top: 24px;
	font-size: $font-body-small;
	line-height: 1.6;
	font-style: italic;

   em {
	font-weight: 600;
   }
}
```

**Review by millerf — DISMISSED** [review]
When loading the page, I have 2 errors in the console that need mending...
HAppens when loading the Earn page (http://calypso.localhost:3000/earn/fieldguide.automattic.com for example)

It does not happen on `trunk`... 



Screenshot 2023-06-01 at 15 22 53

**Comment by edanzer** [comment]
This is testing as expected for me. 
- [x] Created a new newsletter site on a free plan, and checked the new commission notice (10%).
- [x] Upgraded to the Personal plan and checked the commission notice again (8$). 
- [x] Upgraded to the Premium plan and checked again (4%).
- [x] Upgraded to the Business plan and checked again (2%).
- [x] Installed a plugin, which forces the migration to atomic, and then checked again on atomic (2%).

Notes:

I did confirm I can see the same console error @millerf above. Not sure what's generating that as I don't see any nested p tags where that error suggests. 

I'm also seeing a minor TS error in VSCode in the CommissionFees component. It's not liking that we're supplying a number (vs a string) to parseFloat():

ts-error 2

File: client/my-sites/earn/components/commission-fees.tsx (hunk 5 above)
Comment (simison) [reply client/my-sites/earn/components/commission-fees.tsx]: I agree it's not great, but I wanted to use what was already there and translated, and then come back and change it with more time & translations separately:

From `trunk`:

https://github.com/Automattic/wp-calypso/blob/93b98e0fe0e116dc39225a577e51642503bc21c1/client/my-sites/earn/memberships/index.jsx#L114-L120

In docs we say "US$ 0.30" which is clearer, and Cris suggested to just say "Stripe fees". We can also always link to Stripe's fees page.

File: client/my-sites/earn/components/commission-fees.tsx (hunk 6 above)
Comment (simison) [reply client/my-sites/earn/components/commission-fees.tsx:25]: Agreed, gotta switch to `__()`  at least when we redesign the page, but not a blocker now imho.

File: client/my-sites/earn/style.scss (hunk 7 above)
Comment (simison) [reply client/my-sites/earn/style.scss]: For sure! I copied it (and also duplicated some CSS) from earn stats page but figured we have redesign coming up.. :-)

File: client/my-sites/earn/components/commission-fees.tsx (hunk 5 above)
Comment (millerf) [reply client/my-sites/earn/components/commission-fees.tsx]: > We can also always link to Stripe's fees page.

Yes, that sounds like a good idea to me....

File: client/my-sites/earn/components/commission-fees.tsx (hunk 6 above)
Comment (millerf) [reply client/my-sites/earn/components/commission-fees.tsx:25]: 👍

**Comment by millerf** [comment]
@simison The error happens because the CommisionFees is a `<div/>` and is nested inside this `<p>`.
We should transform this `<p/>` to a `<div/>` and it should all work out. 

Screenshot 2023-06-01 at 17 56 41

File: client/my-sites/earn/style.scss (hunk 7 above)
Comment (simison) [reply client/my-sites/earn/style.scss]: fixed!

**Comment by simison** [comment]
Fixed poor looking copy here noticed by @mikemayhem3030 and removed duplicate that I added earlier higher up on the page —

**Before**
Screenshot 2023-06-02 at 12 43 37

**After**
Screenshot 2023-06-02 at 12 38 07

The copy was in code already, but it didn't show up due to bug that @millerf fixed, so didn't notice the copy appeared there.

The copy "No membership fees / No monthly or annual fees charged." is still poor but I'll change it in follow-up PR as it needs a bit more care due to translation wait-time. It refers to not needing paid plan, and there not being minimum fixed fee we charge.

**Comment by simison** [comment]
Fixed react and typing errors https://github.com/Automattic/wp-calypso/pull/77552/commits/7a4b26a28edb5eb9bd59a213806120b0a74156c1  https://github.com/Automattic/wp-calypso/pull/77552/commits/462c995617a5770a7e617f8cf447c292116a8a7e

**Pull request merged**